src/clients/SpokePoolClient.ts

```diff
@@ -283,10 +301,11 @@ export class SpokePoolClient extends BaseAbstractClient {
 
     if (isV2Deposit(deposit)) {
       const v2SpeedUps = this.speedUps[depositor]?.[depositId]?.filter(isV2SpeedUp);
-      const maxSpeedUp = v2SpeedUps?.reduce(
-        (prev, current) => (prev.newRelayerFeePct.gt(current.newRelayerFeePct) ? prev : current),
-        { newRelayerFeePct: deposit.relayerFeePct } as V2SpeedUp
-      );
+      const maxSpeedUp = v2SpeedUps?.reduce((prev, current) => {
+        assert(isV2SpeedUp(prev) && isV2SpeedUp(current)); // tsc hinting.
```

Comment:
Would this not work to convince typescript that these are the input and output types? And if not, what is its complaint?

```ts
const maxSpeedUp = v2SpeedUps?.reduce((prev: V2SpeedUp, current: V2SpeedUp):  V2SpeedUp => {
...
```

Reply:
Resolved here: https://github.com/across-protocol/sdk-v2/pull/501/commits/c3b6cb7cd50ad315a53ff5a642aa0c9dbd910ff4

Thanks for the insights on this!!

Reply:
Further update: https://github.com/across-protocol/sdk-v2/pull/501/commits/3e260c134a0f9e49e0cd986bf3e708e232260c88